### PR TITLE
Finish translation to zh_CN assigned to group 5 (Leo Lee)

### PR DESCRIPTION
**Pull request by Leo-Lee15**

### CODE REVIEW

**Comment by MichaelChirico** [comment]
Great! Please review @fengqifang

File: po/R-zh_CN.po [hunk 1]
```diff
@@ -391,100 +391,112 @@ msgid "i has evaluated to type"
...
 msgid "but there are"
-msgstr ""
+msgstr "但是存在"
 
 msgid ""
 "rows. Recycling of logical i is no longer allowed as it hides more bugs than "
 "is worth the rare convenience. Explicitly use rep(...,length=.N) if you "
 "really need to recycle."
-msgstr ""
+msgstr "行。不再允许布尔值 i 循环补齐，尽管这样有些许便利，但会隐藏更多的 bug，得不偿失。
```
Comment (MichaelChirico) [inline po/R-zh_CN.po]: each line needs to end with `"` and each line needs to begin with `"`

File: po/zh_CN.po [hunk 2]
```diff
@@ -724,205 +724,223 @@ msgstr ""
...
-msgstr ""
+msgstr "length(names)!=length(SD)"
 
 #: dogroups.c:73
 #, c-format
 msgid ""
 "Internal error: size-0 type %d in .SD column %d should have been caught "
 "earlier"
 msgstr ""
+"内部错误：.SD 列中类型 %d size-0 的问题未能被提前捕获到"
```
Comment (MichaelChirico) [inline po/zh_CN.po]: `%d` only appears once in the translation, but twice in the English message

File: po/zh_CN.po [hunk 3]
```diff
@@ -724,205 +724,223 @@ msgstr ""
...
 msgid "length(xknames)!=length(xSD)"
-msgstr ""
+msgstr "length(xknames)!=length(xSD)"
 
 #: dogroups.c:87
 #, c-format
 msgid ""
 "Internal error: type %d in .xSD column %d should have been caught by now"
 msgstr ""
+"内部错误：.xSD 列中类型 %d 的问题未能被提前捕获到"
```
Comment (MichaelChirico) [inline po/zh_CN.po]: Ditto here -- one `%d` in translation, two in original message

File: po/zh_CN.po [hunk 4]
```diff
@@ -724,205 +724,223 @@ msgstr ""
...
+"内部错误：CHARSXP '%s' 的 truelength (%d) 为负。"
+"请将此问题汇报给 data.table 问题追踪器。"
 
 #: chmatch.c:60
 #, c-format
 msgid ""
 "Failed to allocate %lld bytes working memory in chmatchdup: length(table)=%d "
 "length(unique(table))=%d"
 msgstr ""
+"无法为 chmatchdup: length(table)=%d length(unique(table))=%d 分配 %lld 字节的内存空间"
```
Comment (MichaelChirico) [inline po/zh_CN.po]: OK I figured this one out too!!

`msgid` is using `%lld->%d->%d` but `msgstr` is using `%d->%d->%lld`.

It seems pretty clear what's intended to me, but apparently `xgettext` is having trouble processing. We have to number the messages.

Also, the numbering will be `1,2,3` (i.e. the numbering occurs across both types, `%d` and `%lld`).

File: po/zh_CN.po [hunk 5]
```diff
@@ -724,205 +724,223 @@ msgstr ""
...
-msgstr ""
+msgstr "'env' 应该是一个环境"
 
 #: dogroups.c:39
 #, c-format
 msgid ""
 "Internal error: unsupported size-0 type '%s' in column %d of 'by' should "
 "have been caught earlier"
 msgstr ""
+"内部错误：未能被提前捕获到 'by' 中第 %d 列不支持类型 '%s' 且size-0 的问题"
```
Comment (MichaelChirico) [inline po/zh_CN.po]: ditto here

File: po/R-zh_CN.po (hunk 1 above)
Comment (Leo-Lee15) [reply po/R-zh_CN.po]: Thanks for your review. I will fix the problems as soon as possible.

**Comment by MichaelChirico** [comment]
Merged. Thanks to both @caiquanyou @Leo-Lee15 !

**Pull request merged**